dismissInAppNotification support on android.

Royce opened the pull request:
Add .dismissInAppNotification on android.

- Separate hide and show animations.
- Implement a little bit of throttling around successive calls to .showInAppNotification

guyca commented on android/app/src/main/java/com/reactnativenavigation/animation/PeekingAnimator.java:
```diff
@@ -28,33 +26,26 @@ public void animate() {
     }
 
-    private Animator createAnimator(View view) {
+    private Animator createAnimator(View view, boolean show) {
```
This is just nitpicking, but we can avoid the boolean and simply have two public methods:
`createShowAnimator(View view)`
`createHideAnimator(View view)`

Royce commented:
@guyca That method (createAnimator) wasn't really required, so I removed it (and rebased), but I still pass the "show" boolean to the PeekingAnimator ctor.

Alternatively, I could make it more like https://github.com/wix/react-native-navigation/blob/master/android/app/src/main/java/com/reactnativenavigation/animation/VisibilityAnimator.java#L43

varungupta85 commented:
@Royce Thanks for the PR. Would the behavior for `showInAppNotification` be changed a little bit for Android after this PR? Before this PR, the in-app notification would appear and disappear shortly. There was no control to keep it there and dismiss it when I want or dismiss it after a set interval. With the addition of `dismissInAppNotification` on Android, would I be able to keep the in-app notification in the window till I explicitly call the dismiss method?

Royce commented:
I'll be doing a few more PRs soon to address things like that.

On 27 Jan. 2017, 2:52 PM +0800, Varun Gupta <notifications@github.com>, wrote:
>
> @Royce (https://github.com/Royce) Thanks for the PR. Would the behavior for showInAppNotification be changed a little bit for Android after this PR? Before this PR, the in-app notification would appear and disappear shortly. There was no control to keep it there and dismiss it when I want or dismiss it after a set interval. With the addition of dismissInAppNotification on Android, would I be able to keep the in-app notification in the window till I explicitly call the dismiss method?
>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub (https://github.com/wix/react-native-navigation/pull/687#issuecomment-275600790), or mute the thread (https://github.com/notifications/unsubscribe-auth/AABlL5AgGF55_khuMUUIfQ6IVEYWkb7cks5rWZQcgaJpZM4LtCpR).
>
>
>
